sound/soc/sof/ipc4-control.c

```diff
@@ -613,7 +613,11 @@ static int _sof_ipc4_bytes_ext_get(struct snd_sof_control *scontrol,
 	if (data_size > size)
 		return -ENOSPC;
 
-	header.numid = scontrol->comp_id;
+	/*
```

Comment:
@ujfalusi @singalsu is this when you do a get for reading the blob back from the DSP?

Reply:
and oh the link is a SOF commit? is that correct?

Reply:
> @ujfalusi @singalsu is this when you do a get for reading the blob back from the DSP?

This is when user space asking for the binary content.
With IPC3 the numid was hardwired to SOF_CTRL_CMD_BINARY but IPC4 uses the component ID, which is more aligned with the numid.

Afaik it works correctly with the comp_id, but @singalsu is experimenting with some new way and that needs the numid to be SOF_CTRL_CMD_BINARY for some reason, I guess he can explain it here.

In any case, I will update the commit message and/or comment.

Reply:
> and oh the link is a SOF commit? is that correct?

We don't have an issue filed for this and @singalsu only mentioned it in a comment which I tried to link. I'll drop the link tag.

Reply:
@ranj063 The issue is that it's confusing that the get response is not the same as the data that the control was set with. Also, the get response is different from every instance of component. e.g. different instances of IIR return different header despite that if they were set up with the same blob in set.

This header wasn't earlier visible but since UCMv2 chose to use the blob format with the header (#9092) I'm now changing sof-ctl to use same format.